src/earn/poolInfoScreen/WithdrawBottomSheet.tsx

```diff
@@ -0,0 +1,154 @@
+      navigate(Screens.EarnEnterAmount, { pool, mode: 'withdraw' })
+      forwardedRef.current?.close()
+    },
+  }
+  return <ActionCard action={action} />
+}
+
+function ClaimAction({
+  forwardedRef,
+  pool,
+}: {
+  forwardedRef: React.RefObject<BottomSheetModalRefType>
+  pool: EarnPosition
+}) {
+  const { t } = useTranslation()
+  const claimType = pool.dataProps.claimType
+
+  const action: WithdrawAction = {
+    name: 'Claim',
+    title:
+      claimType === 'rewards'
+        ? t('earnFlow.poolInfoScreen.withdrawBottomSheet.claimRewards')
+        : t('earnFlow.poolInfoScreen.withdrawBottomSheet.claimEarnings'),
+    details:
+      claimType === 'rewards'
+        ? t('earnFlow.poolInfoScreen.withdrawBottomSheet.claimRewardsDescription')
+        : t('earnFlow.poolInfoScreen.withdrawBottomSheet.claimEarningsDescription'),
+    iconComponent: Trophy,
+    onPress: () => {
+      AppAnalytics.track(EarnEvents.earn_select_withdraw_type, { type: 'Claim' })
+      navigate(Screens.EarnCollectScreen, { pool }) // TODO (ACT-1389): Confirmation screen for Claim
+      forwardedRef.current?.close()
+    },
+  }
+  return <ActionCard action={action} />
+}
+
+function ExitAction({
+  forwardedRef,
+  pool,
+  hasRewards,
+}: {
+  forwardedRef: React.RefObject<BottomSheetModalRefType>
+  pool: EarnPosition
+  hasRewards: boolean
+}) {
+  const { t } = useTranslation()
+
+  const claimType = pool.dataProps.claimType
+  const details = hasRewards
+    ? claimType === 'rewards'
+      ? t('earnFlow.poolInfoScreen.withdrawBottomSheet.exitWithRewardsDescription')
+      : t('earnFlow.poolInfoScreen.withdrawBottomSheet.exitWithEarningsDescription')
+    : t('earnFlow.poolInfoScreen.withdrawBottomSheet.exitDescription')
+
+  const action: WithdrawAction = {
+    name: 'Exit',
+    title: t('earnFlow.poolInfoScreen.withdrawBottomSheet.exit'),
+    details,
+    iconComponent: Exit,
+    onPress: () => {
+      AppAnalytics.track(EarnEvents.earn_select_withdraw_type, { type: 'Exit' })
+      navigate(Screens.EarnCollectScreen, { pool }) // TODO (ACT-1389): Confirmation screen for Claim & Withdraw
+      forwardedRef.current?.close()
+    },
+  }
+  return <ActionCard action={action} />
+}
+
+export default function WithdrawBottomSheet({
+  forwardedRef,
+  pool,
+  canClaim,
+}: {
+  forwardedRef: RefObject<BottomSheetModalRefType>
+  pool: EarnPosition
+  canClaim: boolean
+}) {
+  const { t } = useTranslation()
+  const canPartialWithdraw = getFeatureGate(StatsigFeatureGates.ALLOW_EARN_PARTIAL_WITHDRAWAL)
+
+  return (
+    <BottomSheet
+      forwardedRef={forwardedRef}
+      title={t('earnFlow.poolInfoScreen.withdrawBottomSheet.title')}
+      titleStyle={styles.bottomSheetTitle}
+      testId={'Earn/WithdrawBottomSheet'}
+    >
+      <View style={styles.actionsContainer}>
+        {canPartialWithdraw && (
+          <PartialWithdrawAction forwardedRef={forwardedRef} pool={pool} hasRewards={canClaim} />
+        )}
```

Comment:
is the gate check needed here? this bottom sheet will never show up if the gate is disabled per changes in EarnPoolInfoScreen right?

Reply:
I think it is needed, in EarnPoolInfoScreen, we show the bottom sheet if `(hasRewards || partialWithdrawalsEnabled)`, so if the user has rewards but the gate is disabled, then the bottom sheet will still be shown with the options to claim rewards or exit.